docs/reference/run.md

```diff
@@ -981,12 +980,11 @@ or two examples of how to pass more parameters to that ENTRYPOINT:
 ## EXPOSE (incoming ports)
```

Comment:
@SpencerBrown I can see why this confused you. It said the same thing several different ways.  Let me know if this is easier to follow. @cpuguy83 please check I didn't bork the technical bits.

--->  https://gist.github.com/moxiegirl/ee4dbfb64f1c0191df51

The following `run` command options work with container networking:

    --expose=[]: Expose a port or a range of ports inside the container,
                These are additional to those exposed by the `EXPOSE` instruction
    -P=false   : Publish all exposed ports to the host interfaces
    -p=[]      : Publish a container᾿s port or a range of ports to the host 
                   format: ip:hostPort:containerPort | ip::containerPort | hostPort:containerPort | containerPort
                   Both hostPort and containerPort can be specified as a range of ports. 
                   When specifying ranges for both, the number of container ports in the range must match the number of host ports in the range. (e.g., `-p 1234-1236:1234-1236/tcp`)
                   (use 'docker port' to see the actual mapping)
    --link=""  : Add link to another container (<name or id>:alias or <name or id>)


With the exception of the `EXPOSE` directive, an image developer hasn't got much
control over networking. The `EXPOSE` instruction defines the initial incoming
ports that provide services. These ports are available to processes inside the
container. An operator can use the `--expose` option to add to the exposed ports.

To expose an container's internal port, an operator can start the container with
the `-P` or `-p` flag or start the client container with `--link`. The exposed
port is accessible on the host and the ports are available to any client
that can reach the host.

The `-P` option publishes all the ports to the host interfaces. Docker binds each
exposed port to a random port on the host. The range of ports are within an
*ephemeral port range* defined by `/proc/sys/net/ipv4/ip_local_port_range`. Use
the `-p` flag to explicitly map a single port or range of ports. 

The port number inside the container (where the service listens) does not need
to match the port number exposed on the outside of the container
(where clients connect). For example, inside the container you might have an HTTP
service listening on port 80 (and so you `EXPOSE 80` in the Dockerfile),
but outside the container the port might be 42800. To find the
mapping between the host ports and the exposed ports, use `docker port`.

If the operator uses `--link` when starting a new client container, then the
client container can access the exposed port via a private networking interface.
 Docker will set some environment variables in the client container to help
indicate which interface and port to use. For more information on linking, see
[the guide on linking container together](/userguide/dockerlinks/ )